staging/src/k8s.io/pod-security-admission/policy/check_procMount.go

```diff
@@ -69,6 +83,9 @@ func procMount_1_0(podMetadata *metav1.ObjectMeta, podSpec *corev1.PodSpec) Chec
 		if container.SecurityContext.ProcMount == nil {
 			return
 		}
+		if relaxForUserNS && relaxPolicyForUserNamespacePod(podSpec) && *container.SecurityContext.ProcMount == corev1.UnmaskedProcMount {
```

Comment:
The conditional relaxation based on `relaxPolicyForUserNamespacePod` (hostUsers=false AND UserNamespacesPodSecurityStandards is opted into) looks reasonable.

I think this can be simplified further to short-circuit the container iteration at the top of procMount, like the other places we relax:
```
	// See KEP-127: https://github.com/kubernetes/enhancements/blob/308ba8d/keps/sig-node/127-user-namespaces/README.md?plain=1#L411-L447
	if relaxPolicyForUserNamespacePod(podSpec) {
		return CheckResult{Allowed: true}
	}
```

---

The next thought is whether we need to rev PSS to 1.31 for this. I *think* 1.30 kubelets are the first version that fails safe (refuses to run the pod) if the pod requests hostUsers=false and the user namespace feature is off / unsupported (done in https://github.com/kubernetes/kubernetes/commit/b2a92406efdf4e46648f1d03b9d1bb9fbb3f48f4).

That means the earliest we could safely drop the need to opt-in via UserNamespacesPodSecurityStandards and *only* consider `hostUsers=false` in the API server in PSS is 1.33 (when the oldest supported kubelet is 1.30). At *that* point, we'd need to rev the policies using `relaxPolicyForUserNamespacePod` to 1.33 versions.

I'm not actually sure what revving to 1.31 here gets us... the new fixtures / integration tests are not checking this is allowed, since they run with UserNamespacesPodSecurityStandards off.

Reply:
I just thought it was idiomatic to make a new version when the policy changed. I can revert to 1.0 if you think it better.

Reply:
one question: if we move the validation up higher, wouldn't we allow an invalid procMount type `corev1.SecurityContext{ProcMount: &invalidString}`? personally, I think leaving it just to be Unmasked is better, but I'm not feeling stubborn about it

Reply:
> I just thought it was idiomatic to make a new version when the policy changed

When we relax without the alpha opt-in (in 1.33, I guess, assuming host namespaces is GA at that point), we'll rev then. In the meantime, just relax at the top.

> if we move the validation up higher, wouldn't we allow an invalid procMount type corev1.SecurityContext{ProcMount: &invalidString}?

API validation ensures sure ProcMount is valid / well-formed, we don't have to double that work here

Reply:
done!